### PR TITLE
Added the ability to create service with Mounts

### PR DESCRIPTION
**Pull request by cloutiertyler**
I've added the ability to create a variable number of mounts to the CreateServiceForm, and have tested the functionality. This branch also includes the Safari fixes.

As is I currently don't do any validation on the inputs here though. There doesn't seem to be a great way to dynamically sizing forms with Semantic-UI (or html generally), but what I've written here ends up working quite nicely. Please let me know if I'm missing a better way.

I don't expect to be able to merge as is, but I'd be happy to integrate your feedback into the branch as soon as possible.

Also, there seems to be a mix of tab characters and white spaces everywhere, so I'm removing the tab characters as edit files, for uniformity.

### CODE REVIEW

**Comment by cloutiertyler** [comment]
I cleaned things up a bit and also added placement constraints to the form, since they follow a similar pattern.

**Comment by tombee** [comment]
This is awesome! Thanks so much.  I've given the code a review and it looks good, I'll give it a spin in a little while. 😀

Great work!

**Comment by cloutiertyler** [comment]
Awesome! I made some UI improvements just now as well.

One thing to look out for on this is that it looks like Safari is honoring the validation, but Chrome is not for some reason. I'm not quite sure why that is. On Safari I get the proper boxes and messages, but in Chrome I only get "Please fill out this field" for the non-Dropdown fields which are required.

**Comment by cloutiertyler** [comment]
I'd like to add support for networks next.

**Review by tombee — APPROVED** [review]

**Comment by tombee** [comment]
@cloutiertyler Just gave this a spin, awesome work!

giphy


2017-01-16-13-25-17_1226x811

File: ui/src/components/services/CreateServiceForm.js [hunk 1]
```diff
@@ -63,92 +124,211 @@ export default class CreateServiceForm extends React.Component {
...
+          {this.state.mountInputKeys.map((key, i) => {
+            return (
+              <List.Item key={key}>
+                <Form.Group widths="equal">
+                  <Form.Select name={"Mounts.MountType-" + i} label={i === 0 && "Type"} options={mountTypes} placeholder="Type" required />
+                  <Form.Input name={"Mounts.Source-" + i} label={i === 0 && "Source"} placeholder="volumename or /host/path" required />
+                  <Form.Input name={"Mounts.Target-" + i} label={i === 0 && "Target"} placeholder="/container/path" required />
+                  <Form.Input name={"Mounts.VolumeDriver-" + i} label={i === 0 && "Volume Driver"} placeholder="local" required />
+                  <Form.Select name={"Mounts.ReadOnly-" + i} label={i === 0 && "Read Only"} options={readOnlyOptions} placeholder="Read Only" required />
+                  <Form.Button type="button" tabIndex={i} width={4} icon="minus" label={i === 0 && "Remove"} basic onClick={this.removeMount}/>
```
Comment (tombee) [inline ui/src/components/services/CreateServiceForm.js:295]: A splash of colour on the add/remove buttons might help call out these buttons :)  Can add that in a future PR though if you were interested in adding the attach network UX.

**Comment by ehazlett** [comment]
nice!

**Pull request merged**